### PR TITLE
BaSP-TG-22: Test Projects: get, getById, post

### PR DESCRIPTION
**Pull request by melissabonillasellanes**
Create tests to verify that the functionality of the get, getById, and post endpoints works correctly in the projects.js file in ./routes

### CODE REVIEW

File: src/test/projects.test.js [hunk 1]
```diff
@@ -1 +1,78 @@
...
+    },
+  ],
+};
+
+describe('GET BYID /project/:id', () => {
+  test('Should return status code 200', async () => {
+    const response = await request(app).get('/projects/63531aaa2b654a3fb77054dd').send();
+    expect(response.status).toBe(200);
+    expect(response.body.error).toBeFalsy();
+    expect(response.body.data).toBeDefined();
```
Comment (nicolasblasco) [inline src/test/projects.test.js:33]: We should also test the message of the body

File: src/test/projects.test.js [hunk 2]
```diff
@@ -1 +1,78 @@
...
+  });
+  test('Should return status code 400', async () => {
+    const response = await request(app).get('/projects/568').send();
+    expect(response.status).toBe(400);
+    expect(response.body.error).toBeTruthy();
```
Comment (nicolasblasco) [inline src/test/projects.test.js:38]: We should also test the message of the body

File: src/test/projects.test.js [hunk 3]
```diff
@@ -1 +1,78 @@
...
+  });
+  test('Should return status code 404', async () => {
+    const response = await request(app).get('/projects/63531244ec6456efd12685ef').send();
+    expect(response.status).toBe(404);
+    expect(response.body.error).toBeTruthy();
```
Comment (nicolasblasco) [inline src/test/projects.test.js:43]: We should also test the message of the body

File: src/test/projects.test.js [hunk 4]
```diff
@@ -1 +1,78 @@
...
+});
+
+describe('POST /project', () => {
+  test('Should create an project', async () => {
+    const response = await request(app).post('/projects').send(mockedProject);
+    // eslint-disable-next-line no-underscore-dangle
+    newIdProject = response.body.data._id;
+    expect(response.status).toBe(201);
+    expect(response.body.error).toBeFalsy();
+    expect(response.body.data).toBeDefined();
```
Comment (nicolasblasco) [inline src/test/projects.test.js:54]: We should also test the message of the body

File: src/test/projects.test.js [hunk 5]
```diff
@@ -1 +1,78 @@
...
+  });
+  test('Should not create an project', async () => {
+    const response = await request(app).post('/projects').send();
+    expect(response.status).toBe(400);
+    expect(response.body.error).toBeTruthy();
+    expect(response.body.data).toBeUndefined();
```
Comment (nicolasblasco) [inline src/test/projects.test.js:60]: We should also test the message of the body

File: src/test/projects.test.js [hunk 6]
```diff
@@ -1 +1,78 @@
...
+  });
+});
+
+describe('GET /project', () => {
+  test('Should return status code 200', async () => {
+    const response = await request(app).get('/projects').send();
+    expect(response.status).toBe(200);
+    expect(response.body.error).toBeFalsy();
+    expect(response.body.data).toBeDefined();
```
Comment (nicolasblasco) [inline src/test/projects.test.js:69]: We should also test the message of the body

File: src/test/projects.test.js [hunk 7]
```diff
@@ -1 +1,78 @@
...
+  });
+  test('Should return status code 404', async () => {
+    await request(app).delete('/projects/63531aaa2b654a3fb77054dd');
+    await request(app).delete(`/projects/${newIdProject}`);
+    const response = await request(app).get('/projects/').send();
+    expect(response.status).toBe(404);
+    expect(response.body.error).toBeTruthy();
```
Comment (nicolasblasco) [inline src/test/projects.test.js:76]: We should also test the message of the body and that the data is empty

**Review by nicolasblasco — CHANGES_REQUESTED** [review]
